runelite-client/src/main/java/net/runelite/client/plugins/questlist/QuestListPlugin.java

```diff
@@ -0,0 +1,401 @@
+		final Widget miniList = client.getWidget(QuestContainer.MINI_QUESTS.widgetInfo);
+
+		if (container == null || freeList == null || memberList == null || miniList == null)
+		{
+			return;
+		}
+
+		updateList(QuestContainer.FREE_QUESTS, filter);
+		updateList(QuestContainer.MEMBER_QUESTS, filter);
+		updateList(QuestContainer.MINI_QUESTS, filter);
+
+		memberList.setOriginalY(freeList.getOriginalY() + freeList.getOriginalHeight() + ENTRY_PADDING);
+		miniList.setOriginalY(memberList.getOriginalY() + memberList.getOriginalHeight() + ENTRY_PADDING);
+
+		freeList.revalidateScroll();
+		memberList.revalidateScroll();
+		miniList.revalidateScroll();
+
+		int y = miniList.getRelativeY() + miniList.getHeight();
+
+		int newHeight = 0;
+		if (container.getScrollHeight() > 0)
+		{
+			newHeight = (container.getScrollY() * y) / container.getScrollHeight();
+		}
+
+		container.setScrollHeight(y);
+		container.revalidateScroll();
+
+		client.runScript(
+			ScriptID.UPDATE_SCROLLBAR,
+			WidgetInfo.QUESTLIST_SCROLLBAR.getId(),
+			WidgetInfo.QUESTLIST_CONTAINER.getId(),
+			newHeight
+		);
+	}
+
+	private void updateList(QuestContainer questContainer, String filter)
+	{
+		Widget list = client.getWidget(questContainer.widgetInfo);
+		if (list == null)
+		{
+			return;
+		}
+
+		Collection<QuestWidget> quests = questSet.get(questContainer);
+
+		if (quests != null)
+		{
+			// Check to make sure the list hasn't been rebuild since we were last her
+			// Do this by making sure the list's dynamic children are the same as when we last saw them
+			if (quests.stream().noneMatch(w ->
+			{
+				Widget codeWidget = w.getQuest();
+				if (codeWidget == null)
+				{
+					return false;
+				}
+				return list.getChild(codeWidget.getIndex()) == codeWidget;
+			}))
+			{
+				quests = null;
+			}
+		}
+
+		if (quests == null)
+		{
+			// Find all of the widgets that we care about, grouping by their Y value
+			Map<Integer, QuestWidget> questMap = new TreeMap<>();
+
+			for (Widget w : list.getDynamicChildren())
+			{
+				String text = w.getText();
+				// Ignore irrelevant widgets
+				if (w.isSelfHidden() || text.equals("Free Quests") || text.equals("Members' Quests") || text.equals("Miniquests"))
+				{
+					continue;
+				}
+
+				QuestWidget c = questMap.computeIfAbsent(w.getRelativeY(), k -> new QuestWidget());
+				c.setQuest(w);
+				c.setTitle(Text.removeTags(w.getText()).toLowerCase());
+			}
+
+			quests = questMap.values();
+			questSet.put(questContainer, quests);
+		}
+
+		// offset because of header
+		int y = 20;
+		for (QuestWidget questInfo : quests)
+		{
+			Widget quest = questInfo.getQuest();
+			QuestState questState = QuestState.getByColor(quest.getTextColor());
+
+			boolean hidden = !(filter.isEmpty()
+				|| questInfo.getTitle().contains(filter))
+				|| (filter.isEmpty() && hideCompleted && questState == QuestState.COMPLETE);
+
+			quest.setHidden(hidden);
+			quest.setOriginalY(y);
```

Comment:
`revalidate()` is required after changing `Original`s

Reply:
In my testing it seemed to work fine only revalidating the parent container. Is this not OK?

Reply:
its working because of the `revalidateScroll`s, but its preferred to have `revalidate`s even if they are unneeded